pkg/logs/message/origin.go

```diff
@@ -67,3 +69,29 @@ func (o *Origin) TagsPayload() []byte {
 func (o *Origin) SetTags(tags []string) {
 	o.tags = tags
 }
+
+// SetSource sets the source of the origin.
+func (o *Origin) SetSource(source string) {
+	o.source = source
+}
+
+// Source returns the source of the origin.
```

Comment:
nit: Yeah, though I find the description to be a bit misleading if you're not looking at the code; it's the source if it has been overridden in the configuration, otherwise it's the source of the message if it has been set.